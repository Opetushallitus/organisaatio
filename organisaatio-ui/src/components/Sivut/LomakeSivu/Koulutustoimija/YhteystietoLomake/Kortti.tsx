import * as React from 'react';
import styles from './YhteystietoLomake.module.css';
import Input from '@opetushallitus/virkailija-ui-components/Input';
import { postinumeroSchema } from '../../../../../ValidationSchemas/YhteystietoLomakeSchema';
import { Control, UseFormSetValue } from 'react-hook-form/dist/types/form';
import { Yhteystiedot } from '../../../../../types/types';
import { useWatch } from 'react-hook-form';
import { useContext } from 'react';
import { KoodistoContext, LanguageContext } from '../../../../../contexts/contexts';
import { FieldErrors } from 'react-hook-form/dist/types/errors';

const postiOsoiteToimipaikkaFiName = 'kieli_fi#1.postiOsoiteToimipaikka';
const postiOsoiteToimipaikkaSvName = 'kieli_sv#1.postiOsoiteToimipaikka';
const kayntiOsoiteToimipaikkaSvName = 'kieli_sv#1.kayntiOsoiteToimipaikka';
const kayntiOsoiteToimipaikkaFiName = 'kieli_fi#1.kayntiOsoiteToimipaikka';

type OsoitteentoimipaikkaProps = {
    name:
        | typeof postiOsoiteToimipaikkaFiName
        | typeof postiOsoiteToimipaikkaSvName
        | typeof kayntiOsoiteToimipaikkaSvName
        | typeof kayntiOsoiteToimipaikkaFiName;
    labelTxt: string;
    control: Control<Yhteystiedot>;
};

const OsoitteenToimipaikkaKentta = ({ name, control }: OsoitteentoimipaikkaProps) => {
    const toimipaikka = useWatch({ control, name });
    return <span className={styles.ToimipaikkaText}>{toimipaikka}</span>;
};

type props = {
    kieli: 'fi' | 'sv' | 'en';
    setYhteystiedotValue: UseFormSetValue<Yhteystiedot>;
    validationErrors: FieldErrors<Yhteystiedot>;
    formControl: Control<Yhteystiedot>;
};
const mapToimipaikkaName = (name: string, kieli: props['kieli']): OsoitteentoimipaikkaProps['name'] =>
    `${kieli}${name.search('kaynti') > -1 ? 'kaynti' : 'posti'}OsoiteToimipaikka` as OsoitteentoimipaikkaProps['name'];

export const Kortti = ({ kieli, setYhteystiedotValue, validationErrors, formControl }: props) => {
    const { i18n } = useContext(LanguageContext);
    const { postinumerotKoodisto } = useContext(KoodistoContext);
    const toimipaikkaOnChange = (e): void => {
        const koodit = postinumerotKoodisto.koodit();
        const { value: postinumero, name } = e.target;
        const toimipaikkaName = mapToimipaikkaName(name, kieli);
        if (postinumeroSchema.required().validate(postinumero)) {
            const postinumeroKoodi = koodit.find((koodi) => koodi.arvo === postinumero);
            if (postinumeroKoodi) {
                const {
                    nimi: { [kieli]: toimipaikka },
                } = postinumeroKoodi;
                setYhteystiedotValue(toimipaikkaName, toimipaikka as string);
            } else setYhteystiedotValue(toimipaikkaName, '');
        } else setYhteystiedotValue(toimipaikkaName, '');
    };
    return (
        <div>
            <div className={styles.Rivi}>
                <div className={styles.Kentta}>
                    <label>{i18n.translate('YHTEYSTIEDOT_POSTIOSOITE')} *</label>
                    <Input name={`${kieli}.postiOsoite`} error={!!validationErrors[kieli]?.postiOsoite} />
                </div>
                <div className={styles.KenttaLyhyt}>
                    <label>{i18n.translate('YHTEYSTIEDOT_POSTINUMERO')}</label>
                    <Input
                        name={`${kieli}.postiOsoitePostiNro`}
                        onChange={toimipaikkaOnChange}
                        error={!!validationErrors[kieli]?.postiOsoitePostiNro}
                    />
                </div>
                <OsoitteenToimipaikkaKentta
                    name={`${kieli}.postiOsoiteToimipaikka` as OsoitteentoimipaikkaProps['name']}
                    labelTxt={i18n.translate('YHTEYSTIEDOT_TOIMIPAIKKA')}
                    control={formControl}
                />
            </div>
        </div>
    );
};
