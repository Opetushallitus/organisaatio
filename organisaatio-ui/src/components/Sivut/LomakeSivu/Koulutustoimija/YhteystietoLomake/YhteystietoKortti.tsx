import * as React from 'react';
import styles from './YhteystietoLomake.module.css';
import Input from '@opetushallitus/virkailija-ui-components/Input';
import Textarea from '@opetushallitus/virkailija-ui-components/Textarea';
import { postinumeroSchema } from '../../../../../ValidationSchemas/YhteystietoLomakeSchema';
import { Control, UseFormRegister, UseFormSetValue } from 'react-hook-form/dist/types/form';
import { KenttaError, Language, Yhteystiedot } from '../../../../../types/types';
import { useWatch } from 'react-hook-form';
import { Kentta, KenttaLyhyt, Rivi } from '../../LomakeFields/LomakeFields';
import { ValidationResult } from 'joi';
import { useAtom } from 'jotai';
import { postinumerotKoodistoAtom } from '../../../../../api/koodisto';
import { languageAtom } from '../../../../../api/lokalisaatio';

const postiOsoiteToimipaikkaFiName = 'fi.postiOsoiteToimipaikka';
const postiOsoiteToimipaikkaSvName = 'sv.postiOsoiteToimipaikka';
const kayntiOsoiteToimipaikkaSvName = 'sv.kayntiOsoiteToimipaikka';
const kayntiOsoiteToimipaikkaFiName = 'fi.kayntiOsoiteToimipaikka';

type OsoitteentoimipaikkaProps = {
    name:
        | typeof postiOsoiteToimipaikkaFiName
        | typeof postiOsoiteToimipaikkaSvName
        | typeof kayntiOsoiteToimipaikkaSvName
        | typeof kayntiOsoiteToimipaikkaFiName;
    labelTxt: string;
    control: Control<Yhteystiedot>;
};

type props = {
    kieli: Language;
    yhteystiedotRegister: UseFormRegister<Yhteystiedot>;
    setYhteystiedotValue: UseFormSetValue<Yhteystiedot>;
    formControl: Control<Yhteystiedot>;
    osoitteetOnEri: boolean;
    validationErrors: ValidationResult;
    readOnly?: boolean;
    isYtj: boolean;
};

const RiviKentta: React.FC<{ error?: KenttaError; label: string; isRequired?: boolean }> = ({
    error,
    label,
    children,
    isRequired = false,
}) => {
    return (
        <Rivi>
            <Kentta isRequired={isRequired} error={error} label={label}>
                {children}
            </Kentta>
        </Rivi>
    );
};

const PostinumeroKentta = ({ children, toimipaikkaName: name, control, label, isRequired = false, error }) => {
    const toimipaikka = useWatch({ control, name });
    return (
        <Rivi>
            <KenttaLyhyt isRequired label={label} error={error}>
                {children}
            </KenttaLyhyt>
            <span className={styles.ToimipaikkaText}>{toimipaikka}</span>
        </Rivi>
    );
};

const OtsikkoRivi = ({ label }) => {
    const [i18n] = useAtom(languageAtom);
    return (
        <div className={styles.EnsimmainenRivi}>
            <h3>{i18n.translate(label)}</h3>
        </div>
    );
};

const getErrorDetails = (validationErrors) => {
    const details = validationErrors?.error?.details;
    if (details && details.length > 0) {
        const {
            path: [lang, name],
            message,
        } = details[0];
        return { lang, name, message };
    } else {
        return { lang: 'fi', name: 'no error', message: undefined };
    }
};

function getError(error: { name: string; lang: string }, kortinKieli: Language, name: string): KenttaError {
    return {
        ref: {
            name: (error.lang === kortinKieli && error.name === name && name) || undefined,
        },
    };
}

export const YhteystietoKortti = ({
    kieli: kortinKieli,
    setYhteystiedotValue,
    validationErrors,
    formControl,
    osoitteetOnEri,
    yhteystiedotRegister,
    readOnly,
    isYtj,
}: props) => {
    const [postinumerotKoodisto] = useAtom(postinumerotKoodistoAtom);
    const ytjReadOnly = isYtj && kortinKieli === 'fi';
    const registerToimipaikkaUpdate = (toimipaikkaName, { onChange: originalOnchange, ...rest }) => {
        const koodit = postinumerotKoodisto.koodit();
        const kieli = toimipaikkaName.substr(toimipaikkaName.indexOf('_') + 1, 2) as 'fi' | 'sv';
        const onChange = (e) => {
            const postinumero = e.target.value;
            if (postinumeroSchema.required().validate(postinumero)) {
                const postinumeroKoodi = koodit.find((koodi) => koodi.arvo === postinumero);
                if (postinumeroKoodi) {
                    const {
                        nimi: { [kieli]: toimipaikka },
                    } = postinumeroKoodi;
                    setYhteystiedotValue(toimipaikkaName, toimipaikka);
                } else setYhteystiedotValue(toimipaikkaName, '');
            } else setYhteystiedotValue(toimipaikkaName, '');
            originalOnchange(e);
        };
        return { onChange, ...rest };
    };
    const error = getErrorDetails(validationErrors);
    if (kortinKieli === 'en')
        return (
            <div key={kortinKieli} className={styles.KorttiKehys}>
                <OtsikkoRivi label={`YHTEYSTIEDOTKORTTI_OTSIKKO_${kortinKieli}`} />
                <RiviKentta
                    label="YHTEYSTIEDOT_POSTIOSOITE_MUU"
                    isRequired
                    error={getError(error, kortinKieli, 'postiOsoite')}
                >
                    <Textarea
                        disabled={readOnly}
                        {...yhteystiedotRegister(`${kortinKieli}.postiOsoite` as const)}
                        error={error.lang === kortinKieli && error.name === 'postiOsoite'}
                    />
                </RiviKentta>
                <RiviKentta label="YHTEYSTIEDOT_PUHELINNUMERO">
                    <Input
                        disabled={readOnly}
                        {...yhteystiedotRegister(`${kortinKieli}.puhelinnumero` as const)}
                        error={getError(error, kortinKieli, 'puhelinnumero')}
                    />
                </RiviKentta>
                <RiviKentta
                    label="YHTEYSTIEDOT_SAHKOPOSTIOSOITE"
                    isRequired
                    error={getError(error, kortinKieli, 'email')}
                >
                    <Input
                        disabled={readOnly}
                        {...yhteystiedotRegister(`${kortinKieli}.email` as const)}
                        error={error.lang === kortinKieli && error.name === 'email'}
                    />
                </RiviKentta>
                <RiviKentta label="YHTEYSTIEDOT_WWW_OSOITE">
                    <Input
                        disabled={readOnly}
                        {...yhteystiedotRegister(`${kortinKieli}.www` as const)}
                        error={error.lang === kortinKieli && error.name === 'www'}
                    />
                </RiviKentta>
            </div>
        );
    return (
        <div key={kortinKieli} className={styles.KorttiKehys}>
            <OtsikkoRivi label={`YHTEYSTIEDOTKORTTI_OTSIKKO_${kortinKieli}`} />
            <RiviKentta label="YHTEYSTIEDOT_POSTIOSOITE" isRequired error={getError(error, kortinKieli, 'postiOsoite')}>
                <Input
                    disabled={readOnly || ytjReadOnly}
                    {...yhteystiedotRegister(`${kortinKieli}.postiOsoite` as const)}
                    error={error.lang === kortinKieli && error.name === 'postiOsoite'}
                />
            </RiviKentta>
            <PostinumeroKentta
                isRequired
                label="YHTEYSTIEDOT_POSTINUMERO"
                toimipaikkaName={`${kortinKieli}.postiOsoiteToimipaikka` as OsoitteentoimipaikkaProps['name']}
                control={formControl}
                error={getError(error, kortinKieli, 'postiOsoitePostiNro')}
            >
                <Input
                    disabled={readOnly || ytjReadOnly}
                    {...registerToimipaikkaUpdate(
                        `${kortinKieli}.postiOsoiteToimipaikka`,
                        yhteystiedotRegister(`${kortinKieli}.postiOsoitePostiNro` as const)
                    )}
                    error={error.lang === kortinKieli && error.name === 'postiOsoitePostiNro'}
                />
            </PostinumeroKentta>
            {osoitteetOnEri && [
                <RiviKentta label="YHTEYSTIEDOT_KAYNTIOSOITE">
                    <Input
                        disabled={readOnly}
                        {...yhteystiedotRegister(`${kortinKieli}.kayntiOsoite` as const)}
                        error={error.lang === kortinKieli && error.name === 'kayntiOsoite'}
                    />
                </RiviKentta>,
                <PostinumeroKentta
                    label="YHTEYSTIEDOT_POSTINUMERO'"
                    toimipaikkaName={`${kortinKieli}.kayntiOsoiteToimipaikka` as OsoitteentoimipaikkaProps['name']}
                    control={formControl}
                    error={getError(error, kortinKieli, 'kayntiOsoitePostiNro')}
                >
                    <Input
                        disabled={readOnly}
                        {...registerToimipaikkaUpdate(
                            `${kortinKieli}.kayntiOsoiteToimipaikka`,
                            yhteystiedotRegister(`${kortinKieli}.kayntiOsoitePostiNro` as const)
                        )}
                        error={error.lang === kortinKieli && error.name === 'kayntiOsoitePostiNro'}
                    />
                </PostinumeroKentta>,
            ]}
            <RiviKentta label="YHTEYSTIEDOT_PUHELINNUMERO">
                <Input
                    disabled={readOnly || ytjReadOnly}
                    {...yhteystiedotRegister(`${kortinKieli}.puhelinnumero` as const)}
                    name={`${kortinKieli}.puhelinnumero`}
                    error={error.lang === kortinKieli && error.name === 'puhelinnumero'}
                />
            </RiviKentta>
            <RiviKentta label="YHTEYSTIEDOT_SAHKOPOSTIOSOITE" isRequired error={getError(error, kortinKieli, 'email')}>
                <Input
                    disabled={readOnly}
                    {...yhteystiedotRegister(`${kortinKieli}.email` as const)}
                    error={error.lang === kortinKieli && error.name === 'email'}
                />
            </RiviKentta>
            <RiviKentta label="YHTEYSTIEDOT_WWW_OSOITE">
                <Input
                    disabled={readOnly || ytjReadOnly}
                    {...yhteystiedotRegister(`${kortinKieli}.www` as const)}
                    error={error.lang === kortinKieli && error.name === 'www'}
                />
            </RiviKentta>
        </div>
    );
};
