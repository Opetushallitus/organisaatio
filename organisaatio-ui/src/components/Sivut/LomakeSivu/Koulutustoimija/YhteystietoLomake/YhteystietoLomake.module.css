.UloinKehys {
    padding: 2rem;
    display: flex;
    flex-direction: column;
    flex-wrap: nowrap;
}

.EnsimmainenRivi {
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
}
.EnsimmainenRivi > h3 {
    font-weight: bolder;
    padding-bottom: 1rem;
    color: #0a789c
}

.Rivi {
    display: flex;
    flex-direction: row;
    align-items: flex-end;
    padding-bottom: 2rem;
    flex-wrap: wrap;
}

.Ruudukko {
    display: grid;
    grid-gap: 1rem;
    grid-template-columns: min-content auto;
    padding-right: 2rem;
}

.Kentta {
    width: 15rem;
    padding-right: 5rem
}

.KenttaLyhyt {
    margin-right: 2rem;
    width: 5rem;
}

.PiilotaNappiKentta {
    margin-left: 1rem;
    margin-right: 5rem;
}


.AvainKevyestiBoldattu {
    font-weight: 700;
}

.ReadOnly {
    font-size: 16px;
    -webkit-appearance: none;
    border: none
}

.Nappi {
    height: 2.5rem;
}

@media (min-width: 900px) {
    .AvainKevyestiBoldattu {
        white-space: nowrap;
    }
}

.ToimipaikkaText {
    padding-bottom: .5rem;
}

.KorttiKehys {
    border: 3px solid #ccc;
    padding: 1rem 1rem 1rem 1.5rem;
    margin: 1rem
}

.KortitContainer {
    display: flex;
    flex-direction: row;
    align-items: flex-end;
    padding-bottom: 2rem;
    flex-wrap: wrap;
}