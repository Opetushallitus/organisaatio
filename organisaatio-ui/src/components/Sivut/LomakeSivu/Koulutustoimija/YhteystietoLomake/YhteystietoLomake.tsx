import * as React from 'react';
import styles from './YhteystietoLomake.module.css';
import Input from "@opetushallitus/virkailija-ui-components/Input";
import Checkbox from "@opetushallitus/virkailija-ui-components/Checkbox";
import RadioGroup from "@opetushallitus/virkailija-ui-components/RadioGroup";
import {useState} from "react";
import {Koodi, Osoite, Yhteystiedot} from "../../../../../types/types";
import useAxios from "axios-hooks";
import Spin from "@opetushallitus/virkailija-ui-components/Spin";
import {SyntheticEvent} from "react";

type yhteystietoProps = {
    yhteystiedot: Yhteystiedot[]
    handleOnChange: ({ name, value }: { name: string; value: any; }) => void
    postinumerot: Koodi[]
}
const urlPrefix = process.env.NODE_ENV === 'development' ? '/api' : '/organisaatio';

export default function YhteystietoLomake(props: yhteystietoProps) {
    const [kieleksi, setKieleksi ] = useState('kieli_fi#1');
    const [postiSamakuinKaynti, setPostiSamakuinKaynti] = useState({ kieleksi, onSama: false });

    const { yhteystiedot, handleOnChange, postinumerot } = props;

    const currentVisibleYhteystiedot = { posti: { osoite: '', postinumeroUri: ''}, kaynti: { osoite: '', postinumeroUri: ''}, puhelin: { numero: ''}, www: { www: ''}, email: { email: ''}};
    yhteystiedot.forEach((yT: any) => {
        if (yT.kieli === kieleksi) {
            if (yT.osoiteTyyppi && yT.osoiteTyyppi=== 'posti') {
                currentVisibleYhteystiedot.posti = yT;
            } else if (yT.osoiteTyyppi && yT.osoiteTyyppi=== 'kaynti') {
                currentVisibleYhteystiedot.kaynti = yT;
            } else if (yT.tyyppi && yT.tyyppi=== 'puhelin') {
                currentVisibleYhteystiedot.puhelin = yT;
            } else if (yT.www) {
                currentVisibleYhteystiedot.www = yT;
            }  else if (yT.email) {
                currentVisibleYhteystiedot.email = yT;
            }
        }
    });
    const handlePostiOsSamaKuinKaynti = (event: SyntheticEvent) => {
        const element = event.target as HTMLInputElement;
        const postiYt = Object.assign({}, yhteystiedot.find((yt: Yhteystiedot) => yt.kieli === kieleksi && yt.osoiteTyyppi === 'posti'));
        const kayntiYt = yhteystiedot.find((yt: Yhteystiedot) => yt.kieli === kieleksi && yt.osoiteTyyppi === 'kaynti');
        if (postiYt && kayntiYt && element.checked) {
            kayntiYt.osoite = postiYt.osoite;
            kayntiYt.postinumeroUri = postiYt.postinumeroUri;
            handleOnChange({ name: 'yhteystiedot', value: yhteystiedot});
        }
        setPostiSamakuinKaynti({kieleksi: kieleksi, onSama: !postiSamakuinKaynti.onSama})
    }
    const handleYhteystietoOnChange = (event: SyntheticEvent) => {
        const element = event.target as HTMLInputElement;
        const name = element.name;
        const oikeankieliset = yhteystiedot.filter((yt: Yhteystiedot) => yt.kieli === kieleksi);
        if( oikeankieliset.length > 0) {
            if (name === 'www' || name === 'email' || name === 'numero') {
                const oikea = oikeankieliset.find(yt => yt.hasOwnProperty(name));
                if(oikea) oikea[name] = element.value;
                else yhteystiedot.push({ kieli: kieleksi, [name]: element.value, })
            } else {
                const jaettu = name.split('.');
                const pohja = jaettu[0] as keyof Yhteystiedot;
                const avain = jaettu[1] as keyof Osoite;
                const oikea = oikeankieliset.find(yt => yt.osoiteTyyppi && yt.osoiteTyyppi === pohja);
                if(oikea) {
                    oikea[avain] = element.value;
                    if (oikea.osoiteTyyppi === 'posti' && kieleksi === postiSamakuinKaynti.kieleksi && postiSamakuinKaynti.onSama) {
                        const kayntiYt = yhteystiedot.find((yt: Yhteystiedot) => yt.kieli === kieleksi && yt.osoiteTyyppi === 'kaynti');
                        if (kayntiYt) {
                            kayntiYt.osoite = oikea.osoite;
                            kayntiYt.postinumeroUri = oikea.postinumeroUri;
                        }
                    }
                }
            }
        }
        handleOnChange({ name: 'yhteystiedot', value: yhteystiedot});
    };
    if (!postinumerot) {
       return <Spin />;
    }
    return(
        <div className={styles.UloinKehys}>
            <div className={styles.Rivi}>
                <div className={styles.Kentta}>
                    <RadioGroup
                        value={kieleksi}
                        options={[
                            { value: 'kieli_fi#1', label: 'Suomeksi' },
                            { value: 'kieli_sv#1', label: 'Ruotsiksi' },
                            { value: 'kieli_en#1', label: 'Englanniksi' },
                        ]}
                        onChange={e => setKieleksi(e.target.value)}
                    />
                </div>
            </div>
            <div className={styles.Rivi}>
                <Checkbox
                    checked={kieleksi === postiSamakuinKaynti.kieleksi && postiSamakuinKaynti.onSama}
                    onChange={handlePostiOsSamaKuinKaynti}
                >Postiosoite on sama kuin käyntiosoite</Checkbox>
            </div>
            <div className={styles.Rivi}>
                <div className={styles.Kentta}>
                    <label>Postiosoite</label>
                    <Input
                        name="posti.osoite"
                        onChange={handleYhteystietoOnChange}
                        value={currentVisibleYhteystiedot.posti && currentVisibleYhteystiedot.posti.osoite}
                    />
                </div>
                <div className={styles.KenttaLyhyt}>
                    <label>Postinumero</label>
                    <Input
                        name="posti.postinumeroUri"
                        onChange={handleYhteystietoOnChange}
                        value={currentVisibleYhteystiedot.posti && currentVisibleYhteystiedot.posti.postinumeroUri}
                    />
                </div>
            </div>
            <div className={styles.Rivi}>
                <div className={styles.Kentta}>
                    <label>Käyntiosoite</label>
                    <Input
                        disabled={kieleksi === postiSamakuinKaynti.kieleksi && postiSamakuinKaynti.onSama}
                        name="kaynti.osoite"
                        onChange={handleYhteystietoOnChange}
                        value={currentVisibleYhteystiedot.kaynti && currentVisibleYhteystiedot.kaynti.osoite}
                    />
                </div>
                <div className={styles.KenttaLyhyt}>
                    <label>Postinumero</label>
                    <Input
                        disabled={kieleksi === postiSamakuinKaynti.kieleksi && postiSamakuinKaynti.onSama}
                        onChange={handleYhteystietoOnChange}
                        value={currentVisibleYhteystiedot.kaynti && currentVisibleYhteystiedot.kaynti.postinumeroUri}
                        name="kaynti.postinumeroUri"
                    />
                </div>
            </div>
            <div className={styles.Rivi}>
                <div className={styles.Kentta}>
                    <label>Puhelinnumero</label>
                    <Input
                        value={currentVisibleYhteystiedot.puhelin && currentVisibleYhteystiedot.puhelin.numero}
                        name="numero"
                        onChange={handleYhteystietoOnChange}
                    />
                </div>
            </div>
            <div className={styles.Rivi}>
                <div className={styles.Kentta}>
                    <label>Sähköpostiosoite</label>
                    <Input
                        value={currentVisibleYhteystiedot.email && currentVisibleYhteystiedot.email.email}
                        name="email"
                        onChange={handleYhteystietoOnChange}
                    />
                </div>
            </div>
            <div className={styles.Rivi}>
                <div className={styles.Kentta}>
                    <label>Www-osoite</label>
                    <Input
                        value={currentVisibleYhteystiedot.www && currentVisibleYhteystiedot.www.www}
                        name="www"
                        onChange={handleYhteystietoOnChange}
                    />
                </div>
            </div>
        </div>
    );
}