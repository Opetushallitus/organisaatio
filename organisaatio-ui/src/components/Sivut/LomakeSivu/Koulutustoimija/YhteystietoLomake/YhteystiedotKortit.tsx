import * as React from 'react';
import { Kortti } from './Kortti';

export const YhteystiedotKortit = ({ osoitteet }) => {
    const cardsArray = osoitteet.map((robot) => <Kortti name={robot.name} email={robot.email} id={robot.id} />);
    return <div>{cardsArray}</div>;
};
