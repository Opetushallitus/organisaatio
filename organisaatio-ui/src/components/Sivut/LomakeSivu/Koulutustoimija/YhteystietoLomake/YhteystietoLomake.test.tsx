import React from 'react';
import { shallow } from 'enzyme';

import YhteystietoLomake, { Props } from './YhteystietoLomake';

const MINIMAL_PROPS: Props = {
    yhteystiedot: [],
    handleOnChange: jest.fn,
};

describe('YhteystietoLomake', () => {
    it('Renders without crashing', () => {
        const element = shallow(<YhteystietoLomake {...MINIMAL_PROPS} />);
        expect(element).toMatchSnapshot();
    });
});
