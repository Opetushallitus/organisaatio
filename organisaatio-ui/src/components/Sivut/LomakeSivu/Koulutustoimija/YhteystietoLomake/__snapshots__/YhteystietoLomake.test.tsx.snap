// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YhteystietoLomake Change lang creates needed yhteystietos 1`] = `
Array [
  Object {
    "kieli": "kieli_fi#1",
    "osoite": "",
    "osoiteTyyppi": "posti",
    "postinumeroUri": "",
    "postitoimipaikka": "",
  },
  Object {
    "kieli": "kieli_fi#1",
    "osoite": "",
    "osoiteTyyppi": "kaynti",
    "postinumeroUri": "",
    "postitoimipaikka": "",
  },
  Object {
    "kieli": "kieli_fi#1",
    "numero": "",
  },
  Object {
    "email": "",
    "kieli": "kieli_fi#1",
  },
  Object {
    "kieli": "kieli_fi#1",
    "www": "",
  },
  Object {
    "kieli": "test",
    "osoite": "",
    "osoiteTyyppi": "posti",
    "postinumeroUri": "",
    "postitoimipaikka": "",
  },
  Object {
    "kieli": "test",
    "osoite": "",
    "osoiteTyyppi": "kaynti",
    "postinumeroUri": "",
    "postitoimipaikka": "",
  },
  Object {
    "kieli": "test",
    "numero": "",
  },
  Object {
    "email": "",
    "kieli": "test",
  },
  Object {
    "kieli": "test",
    "www": "",
  },
]
`;

exports[`YhteystietoLomake Checkbox Copies posti to kaynti when checked 1`] = `
Array [
  Array [
    Object {
      "name": "yhteystiedot",
      "value": Array [
        Object {
          "kieli": "kieli_fi#1",
          "osoite": "testi",
          "osoiteTyyppi": "posti",
        },
        Object {
          "kieli": "kieli_fi#1",
          "osoite": "testi",
          "osoiteTyyppi": "kaynti",
          "postinumeroUri": undefined,
          "postitoimipaikka": "",
        },
        Object {
          "kieli": "kieli_fi#1",
          "numero": "",
        },
        Object {
          "email": "",
          "kieli": "kieli_fi#1",
        },
        Object {
          "kieli": "kieli_fi#1",
          "www": "",
        },
      ],
    },
  ],
]
`;

exports[`YhteystietoLomake Rendering Renders without crashing 1`] = `
<div
  className="UloinKehys"
>
  <div
    className="Rivi"
  >
    <div
      className="Kentta"
    >
      <Component
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "Suomeksi",
              "value": "kieli_fi#1",
            },
            Object {
              "label": "Ruotsiksi",
              "value": "kieli_sv#1",
            },
            Object {
              "label": "Englanniksi",
              "value": "kieli_en#1",
            },
          ]
        }
        value="kieli_fi#1"
      />
    </div>
  </div>
  <div
    className="Rivi"
  >
    <ForwardRef
      checked={false}
      id="checkbox"
      onChange={[Function]}
    >
      YHTEYSTIEDOT_POSTIOSOITE_ON_SAMA_KUIN_KAYNTIOSOITE
    </ForwardRef>
  </div>
  <div
    className="Rivi"
  >
    <div
      className="Kentta"
    >
      <label>
        YHTEYSTIEDOT_POSTIOSOITE
      </label>
      <ForwardRef
        name="posti.osoite"
        onChange={[Function]}
        value=""
      />
    </div>
    <div
      className="KenttaLyhyt"
    >
      <label>
        YHTEYSTIEDOT_POSTINUMERO
      </label>
      <ForwardRef
        name="posti.postinumeroUri"
        onChange={[Function]}
        value=""
      />
    </div>
  </div>
  <div
    className="Rivi"
  >
    <div
      className="Kentta"
    >
      <label>
        YHTEYSTIEDOT_KAYNTIOSOITE
      </label>
      <ForwardRef
        disabled={false}
        name="kaynti.osoite"
        onChange={[Function]}
        value=""
      />
    </div>
    <div
      className="KenttaLyhyt"
    >
      <label>
        YHTEYSTIEDOT_POSTINUMERO
      </label>
      <ForwardRef
        disabled={false}
        name="kaynti.postinumeroUri"
        onChange={[Function]}
        value=""
      />
    </div>
  </div>
  <div
    className="Rivi"
  >
    <div
      className="Kentta"
    >
      <label>
        YHTEYSTIEDOT_PUHELINNUMERO
      </label>
      <ForwardRef
        name="numero"
        onChange={[Function]}
        value=""
      />
    </div>
  </div>
  <div
    className="Rivi"
  >
    <div
      className="Kentta"
    >
      <label>
        YHTEYSTIEDOT_SAHKOPOSTIOSOITE
      </label>
      <ForwardRef
        name="email"
        onChange={[Function]}
        value=""
      />
    </div>
  </div>
  <div
    className="Rivi"
  >
    <div
      className="Kentta"
    >
      <label>
        YHTEYSTIEDOT_WWW_OSOITE
      </label>
      <ForwardRef
        name="www"
        onChange={[Function]}
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`YhteystietoLomake Update address kaynti 1`] = `
Array [
  Array [
    Object {
      "name": "yhteystiedot",
      "value": Array [
        Object {
          "kieli": "kieli_fi#1",
          "osoite": "",
          "osoiteTyyppi": "posti",
          "postinumeroUri": "",
          "postitoimipaikka": "",
        },
        Object {
          "kieli": "kieli_fi#1",
          "osoite": "test",
          "osoiteTyyppi": "kaynti",
          "postinumeroUri": "",
          "postitoimipaikka": "",
        },
        Object {
          "kieli": "kieli_fi#1",
          "numero": "",
        },
        Object {
          "email": "",
          "kieli": "kieli_fi#1",
        },
        Object {
          "kieli": "kieli_fi#1",
          "www": "",
        },
      ],
    },
  ],
]
`;

exports[`YhteystietoLomake Update address posti 1`] = `
Array [
  Array [
    Object {
      "name": "yhteystiedot",
      "value": Array [
        Object {
          "kieli": "kieli_fi#1",
          "osoite": "test",
          "osoiteTyyppi": "posti",
          "postinumeroUri": "",
          "postitoimipaikka": "",
        },
        Object {
          "kieli": "kieli_fi#1",
          "osoite": "",
          "osoiteTyyppi": "kaynti",
          "postinumeroUri": "",
          "postitoimipaikka": "",
        },
        Object {
          "kieli": "kieli_fi#1",
          "numero": "",
        },
        Object {
          "email": "",
          "kieli": "kieli_fi#1",
        },
        Object {
          "kieli": "kieli_fi#1",
          "www": "",
        },
      ],
    },
  ],
]
`;

exports[`YhteystietoLomake Update field email 1`] = `
Array [
  Array [
    Object {
      "name": "yhteystiedot",
      "value": Array [
        Object {
          "kieli": "kieli_fi#1",
          "osoite": "",
          "osoiteTyyppi": "posti",
          "postinumeroUri": "",
          "postitoimipaikka": "",
        },
        Object {
          "kieli": "kieli_fi#1",
          "osoite": "",
          "osoiteTyyppi": "kaynti",
          "postinumeroUri": "",
          "postitoimipaikka": "",
        },
        Object {
          "kieli": "kieli_fi#1",
          "numero": "",
        },
        Object {
          "email": "test",
          "kieli": "kieli_fi#1",
        },
        Object {
          "kieli": "kieli_fi#1",
          "www": "",
        },
      ],
    },
  ],
]
`;

exports[`YhteystietoLomake Update field numero 1`] = `
Array [
  Array [
    Object {
      "name": "yhteystiedot",
      "value": Array [
        Object {
          "kieli": "kieli_fi#1",
          "osoite": "",
          "osoiteTyyppi": "posti",
          "postinumeroUri": "",
          "postitoimipaikka": "",
        },
        Object {
          "kieli": "kieli_fi#1",
          "osoite": "",
          "osoiteTyyppi": "kaynti",
          "postinumeroUri": "",
          "postitoimipaikka": "",
        },
        Object {
          "kieli": "kieli_fi#1",
          "numero": "test",
        },
        Object {
          "email": "",
          "kieli": "kieli_fi#1",
        },
        Object {
          "kieli": "kieli_fi#1",
          "www": "",
        },
      ],
    },
  ],
]
`;

exports[`YhteystietoLomake Update field www 1`] = `
Array [
  Array [
    Object {
      "name": "yhteystiedot",
      "value": Array [
        Object {
          "kieli": "kieli_fi#1",
          "osoite": "",
          "osoiteTyyppi": "posti",
          "postinumeroUri": "",
          "postitoimipaikka": "",
        },
        Object {
          "kieli": "kieli_fi#1",
          "osoite": "",
          "osoiteTyyppi": "kaynti",
          "postinumeroUri": "",
          "postitoimipaikka": "",
        },
        Object {
          "kieli": "kieli_fi#1",
          "numero": "",
        },
        Object {
          "email": "",
          "kieli": "kieli_fi#1",
        },
        Object {
          "kieli": "kieli_fi#1",
          "www": "test",
        },
      ],
    },
  ],
]
`;
